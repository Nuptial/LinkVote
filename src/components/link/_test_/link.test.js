import React from "react";
import ReactDOM from "react-dom";
import Link from "../link";

import { render, cleanup } from "@testing-library/react";

import renderer from "react-test-renderer";

afterEach(cleanup);

const link = {
  name: "testLink",
  url: "http://testalican.com",
  points: 0,
};

it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(<Link link={link}></Link>, div);
});

it("renders link correctly", () => {
  const { getByTestId } = render(
    <Link link={link}></Link>
  );

  expect(getByTestId("link-points")).toHaveTextContent("0");
  expect(getByTestId("link-name")).toHaveTextContent("testLink");
  expect(getByTestId("link-url")).toHaveTextContent("http://testalican.com");
});

it("matches snapshot", () => {
  const tree = renderer
    .create(<Link link={link}></Link>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
