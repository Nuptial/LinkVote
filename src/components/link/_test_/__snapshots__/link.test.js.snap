// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="link-container"
>
  <div
    className="link-points"
  >
    <span
      className="points"
      data-testid="link-points"
    >
      0
    </span>
    POINTS
  </div>
  <div
    className="link-details"
  >
    <span
      className="link-title"
      data-testid="link-name"
    >
      testLink
    </span>
    <span
      className="link-url"
      data-testid="link-url"
    >
      (
      http://testalican.com
      )
    </span>
    <div
      className="link-vote-container"
    >
      <div
        className="link-vote"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-arrow-up fa-w-14 icon"
          data-icon="arrow-up"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Up Vote
      </div>
      <div
        className="link-vote"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-arrow-down fa-w-14 icon"
          data-icon="arrow-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Down Vote
      </div>
    </div>
  </div>
  <img
    alt="delete-icon"
    className="delete-icon"
    src="sign-delete-icon.png"
  />
</div>
`;
